app/models/loan.rb

```diff
@@ -1,9 +1,22 @@
 ## Loan model
 class Loan < ApplicationRecord
   include LoanManager
+  include PayoffManager
 
   belongs_to :deal
-
+  has_many :payoffs
+  has_many :loan_adjustments
+  has_many :construction_draws
+  has_many :invoices
+  has_many :processed_construction_draws, -> { where status: 'processed' },
```

Comment:
custom associations are cool, might be cleaner to use a scope and chain it.  Could use context we should discuss.

Reply:
Yup, I'd like to get your thoughts. My intent usually is to use processed_construction_draws to be able to sum the amounts. 
Instead I could do this in a virtual attribute on the loan
something like:
`def sum_processed_construction_draws
     processed_construction_draws.sum(:amount)
end`

Reply:
^ this might be good - really depends on what is happening on the backend.  How many of these things are getting loaded, how large is the payload.  Is it better to do the math somewhere else?